master/buildbot/www/hooks/gitlab.py

```diff
@@ -108,6 +108,19 @@ def _process_merge_request_change(self, payload, project, event, codebase=None):
         when_timestamp = dateparse(commit['timestamp'])
         # @todo provide and document a way to choose between http and ssh url
         repo_url = attrs['source']['git_http_url']
+
+        # Filter out uninteresting events
+        state = attrs['state']
+        if re.match('^(closed|merged|approved)$', state):
+            log.msg("GitLab MR#{}: Ignoring because state is {}".format(attrs['iid'], state))
+            return []
+        # I'd like to be able to let pushes trigger a retry, but don't know how to do that.
+        # So only allow opens or reopens of an MR to trigger a try build.
+        action = attrs['action']
+        if not re.match('^(open|reopen)$', action):
```

Comment:
I think indeed, we should rebuild in case of merge request update. I think the action for that is just "update": https://gitlab.com/gitlab-org/gitlab-ce/blob/master/app/models/event.rb#L20